docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst

```diff
@@ -308,101 +309,174 @@ Here is a sample of this configuration file:
+
+Build farm configuration sections, e.g. ``awesome_build_farm``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this file, you can specify as many build farm configuration sections as you want,
+each with a header like ``awesome_build_farm`` (i.e. a nice, short name
+you made up). Such a section must contain the following fields:
+
+``build_farm_type``
+"""""""""""""""""""""
+
+This field maps the build farm configuration to a specific build farm class name
+(or in other words, a build farm definition). By default, build farm classes can
+be found in :gh-file-ref:`deploy/buildtools/buildfarm.py`. However, you can specify
+your own custom build farm classes by adding your python file to the ``PYTHONPATH``.
+For example, to use the ``AWSEC2`` build farm class, you would write ``build_farm_type: AWSEC2``.
+
+``args``
+"""""""""""""""""""""
+
+This section specifies all arguments needed for the specific ``build_farm_type`` used.
+For a list of arguments needed for a specific build farm class, users should refer to
+the ``_parse_args`` function in the build farm class given by ``build_farm_type``.
+
+Mainline build farm configurations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are two mainline build farm configurations provided (with their corresponding build farm classes in :gh-file-ref:`deploy/buildtools/buildfarm.py`).
+
+``ec2_build_farm (AWSEC2)``
+"""""""""""""""""""""""""""
+
+.. literalinclude:: /../deploy/sample-backup-configs/sample_config_build_farm.yaml
+   :language: yaml
+   :start-after: DOCREF START: EC2 Build Farm
+   :end-before: DOCREF END: EC2 Build Farm
+
+This build farm configuration provides the default EC2 build farm functionality
+of FireSim 1.13.x and earlier. For each AGFI (FireSim AWS EC2 F1 FPGA Image) being
+built, a new EC2 instance (build host) will be spawned according to the build farm configuration
+arguments. Its ``build_farm_type`` is set as the ``AWSEC2`` build farm class.
+
+The ``instance_type`` argument defines the type of instance that the build will run on.
+Generally, running on a ``z1d.2xlarge`` is sufficient. In our experience, using more
+powerful instances than this provides little gain.
+
+The ``build_instance_market`` argument specifies the AWS market used. You can specify either
+``spot`` or ``ondemand`` here.
+
+When ``build_instance_market: spot``, the ``spot_interruption_behavior`` and ``spot_max_price``
+arguments determines spot instance behaviors. The ``spot_interruption_behavior`` value
+determines what happens to an instance if it receives the interruption signal from AWS. You can specify
+either ``hibernate``, ``stop``, or ``terminate`` for this value. The ``spot_max_price``
+value determines the max price you are willing to pay per instance, in dollars.
+You can also set it to ``ondemand`` to set your max to the on-demand price for the instance.
+
+The ``default_build_dir`` argument specifies the default location of the build directory used
```

Comment:
Can you explain why "default" is included in the name? It suggests there may be reasons it is put elsewhere.

Reply:
Technically, the "default_" part is not needed. The build directory on the build host will always be located by this argument. I added "default_" here because Sagar wanted the names to be consistent between the EC2/Localhost build farms.

Reply:
I added an extra sentence mentioning that this can't be overridden.